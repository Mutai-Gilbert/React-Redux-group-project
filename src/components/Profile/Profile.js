import { useSelector } from 'react-redux';
import classes from './Profile.module.css';

const Profile = () => {
  const missions = useSelector((state) => state.mission.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved !== false);

  let missionContent = <p>No missions reserved</p>;
  if (reservedMissions.length > 0) {
    missionContent = (
      <table className={classes.missions_table}>
        <tbody>
          {reservedMissions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
  return (
    <section className={classes.reserverd_section}>
      <div className={classes.reserved_missions}>
        <h2>My Missions</h2>
        {missionContent}
      </div>
    </section>
  );
};
export default Profile;
