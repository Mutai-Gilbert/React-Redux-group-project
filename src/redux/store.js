import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from '../redux/rockets/rocketsSlice';


const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    mission
  },
});

export default store;
