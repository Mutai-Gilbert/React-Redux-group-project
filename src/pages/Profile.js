import Profile from '../components/Profile/Profile';

const ProfilePage = () => (
  <div>
    <Profile />
  </div>
);
export default ProfilePage;
