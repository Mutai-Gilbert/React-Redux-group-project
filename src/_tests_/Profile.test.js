import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../components/Profile/Profile';
import store from '../redux/store';

describe('Test Profile components', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
