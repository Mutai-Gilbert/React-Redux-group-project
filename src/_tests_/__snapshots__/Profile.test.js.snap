// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Profile components renders correctly 1`] = `
<section
  className="reserverd_section"
>
  <div
    className="reserved_missions"
  >
    <h2>
      My Missions
    </h2>
    <p>
      No missions reserved
    </p>
  </div>
</section>
`;
